src/lib/opencv.ts

```diff
@@ -0,0 +1,158 @@
+import {RemoteLib} from './remote_lib';
+import Docker from 'dockerode';
+import fs, {createWriteStream} from 'fs';
+import {randomUUID} from 'crypto';
+import got from 'got';
+import stream from 'stream';
+import path from 'path';
+
+const OPENCV_IMAGE = 'hdgigante/python-opencv:4.9.0-alpine';
+const CACHE_FOLDER = 'cache';
+
+export class ImageCache {
+  generateIdentifier(): string {
+    return randomUUID() + '.jpg';
+  }
+
+  getCachePath(key: string): string {
+    return `${CACHE_FOLDER}/${key}`;
+  }
+}
+
+interface FileArg {
+  fileType: string,
+  fileUrl: string,
+}
+
+function isFileArg(arg: any): arg is FileArg {
+  return 'object' === typeof arg && null !== arg && 'fileType' in arg;
+}
+
+
+const imageCache = new ImageCache();
+
+class EchoStream extends stream.Writable {
+  private content = '';
+  _write(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null) => void): void {
+    // eslint-disable-next-line
+    this.content += chunk.toString();
+    callback();
+  }
+
+  getContent(): string {
+    return this.content;
+  }
+}
+
+export class OpenCvLib extends RemoteLib {
+  private dockerInstance: Docker|null = null;
+
+  async executeRemoteCall(callName: string, args: any[]): Promise<any> {
+    if (!this.dockerInstance) {
+      this.initDockerInstance();
+    }
+
+    const dockerInstance = this.dockerInstance as Docker;
+
+    const images = await dockerInstance.listImages();
+    if (!images.find(image => image.RepoTags && image.RepoTags.includes(OPENCV_IMAGE))) {
+      await new Promise(resolve => {
+        void dockerInstance.pull(OPENCV_IMAGE, (err: any, stream: NodeJS.ReadableStream) => {
+          dockerInstance.modem.followProgress(stream, resolve);
+        });
+      });
+    }
+
+    const resultImageName = imageCache.generateIdentifier();
+
+    const program = `import cv2
+result = cv2.${callName}(${await this.formatArguments(args)})
```

Comment:
Good point, I've added sanitization to call name and arguments

Reply:
I am not fully confident you handled all cases. There should be a way in python to call a function while sanitizing a set of args and that you could put directly in your code. A kind of `call(functionName, args)`.

Reply:
I don't really see how it would make the code safer to run something like `call(functionName, args)` if call is a Python function VS the code I have written?